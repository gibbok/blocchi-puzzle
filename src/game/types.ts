export enum DirectionEnum {
  N = 'N',
  E = 'E',
  S = 'S',
  W = 'W',
}

export const NO = DirectionEnum.N;
export const ES = DirectionEnum.E;
export const SO = DirectionEnum.S;
export const WE = DirectionEnum.W;

export enum TetroEnum {
  Z = 'Z',
  S = 'S',
  J = 'J',
  T = 'T',
  I = 'I',
  L = 'L',
  O = 'O',
}

export enum NoTetroEnum {
  NoTetro,
}
export const NoTetro = NoTetroEnum.NoTetro;

export type Tile = NoTetroEnum.NoTetro | TetroEnum;

export type TetroPieces = Record<TetroEnum, Record<DirectionEnum, Tetro>>;

export type TetroRow = readonly Tile[];

// The tetromino, a geometric shape composed of four squares/tiles
export type Tetro = readonly TetroRow[];

export type TetroDef = Readonly<{
  type: TetroEnum;
  direction: DirectionEnum;
  x: number;
  y: number;
}>;

export const Z = TetroEnum.Z;
export const S = TetroEnum.S;
export const J = TetroEnum.J;
export const T = TetroEnum.T;
export const I = TetroEnum.I;
export const L = TetroEnum.L;
export const O = TetroEnum.O;

export type Position = number;

export type BoardRow = readonly Tile[];

export type BoardMutable = Tile[][];

// This is the space on which the game is played.
// Tetriminos fall from the top of the Board, and then players turn and move them into their desired places at the bottom.
export type Board = readonly BoardRow[];

export enum ScreenEnum {
  Intro = 'Intro',
  Game = 'Game',
  Over = 'Over',
}

export type InternalState = Readonly<{
  board: Board;
  score: number; // some point based on lines done
  level: number; // based on score, every 1000 score you go to next level, speed is faster
  lines: number; // number of lines done
  currentTetro: TetroDef;
  nextTetro: TetroDef;
  isPlay: boolean;
  isGameOver: boolean;
  screen: ScreenEnum;
}>;

export type PublicState = Omit<InternalState, 'currentTetro' | 'isPlay' | 'isGameOver'>;

export enum KeyEnum {
  Esc = 27,
  Space = 32,
  Left = 37,
  Up = 38,
  Right = 39,
  Down = 40,
}

export type CallBack = () => void;

export type Action = Readonly<{
  type: string;
}>;

export type TransitionState = 'entering' | 'entered' | 'exiting' | 'exited';
export interface TransitionStyles {
  entering?: React.CSSProperties;
  entered?: React.CSSProperties;
  exiting?: React.CSSProperties;
  exited?: React.CSSProperties;
}
