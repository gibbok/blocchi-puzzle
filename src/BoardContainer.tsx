import * as React from 'react';
import { InternalState } from './types';
import { connect } from 'react-redux';

const Board = (state: InternalState) => <div>{JSON.stringify(state)}</div>;

const mapStateToProps = (state: InternalState) => state;

export const BoardContainer = connect(mapStateToProps)(Board);
