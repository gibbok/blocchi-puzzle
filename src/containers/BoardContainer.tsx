import { Board as BoardType, InternalState } from '~game/types';
import { connect } from 'react-redux';
import { mkPublicState } from '~store/reducer';
import { Board } from '~components';

const mapStateToProps = (state: InternalState): { board: BoardType } => {
  const { board } = mkPublicState(state);
  return { board };
};

export const BoardContainer = connect(mapStateToProps)(Board);
