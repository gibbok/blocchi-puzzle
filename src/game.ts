export const calcScore = (rowCompleted: number) => rowCompleted * 100;

export const calcLevel = (score: number) => Math.ceil((score + 1) / 1000);
