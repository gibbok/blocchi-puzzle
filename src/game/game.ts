/**
 * Calculate the game score.
 * @param rowCompleted Number of rows completed
 */
export const calcScore = (rowCompleted: number): number => rowCompleted * 100;

/**
 * Calculate the game level.
 * @param score Game score
 */
export const calcLevel = (score: number): number => Math.ceil((score + 1) / 500);
