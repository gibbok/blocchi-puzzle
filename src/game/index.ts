export * from './board';
export * from './game';
export * from './interaction';
export * from './pieces';
export * from './tetromino';
export * from './types';
