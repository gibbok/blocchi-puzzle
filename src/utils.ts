import { Tetro, Board } from './types';

export const logger = (x: Tetro | Board) => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
