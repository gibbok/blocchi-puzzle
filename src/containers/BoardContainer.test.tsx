import { mapStateToProps, BoardContainer } from './BoardContainer';
import { TetroEnum, I } from '../game/types';
import { mkInitialState } from '../store';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import { BOARD_ROW_EMPTY } from '../utils';

describe('<BoardContainer />', () => {
  it('should mapStateToProps correctly', () => {
    const input = mkInitialState(TetroEnum.I, TetroEnum.J);
    const output = mapStateToProps(input);
    const result = {
      board: [
        [0, 0, 0, 0, I, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, I, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, I, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, I, 0, 0, 0, 0, 0],
        ...Array(16).fill(BOARD_ROW_EMPTY),
      ],
    };

    expect(output).toMatchObject(result);
  });

  it('should render correctly', () => {
    const mockStore = configureMockStore();
    const initialState = mkInitialState(TetroEnum.I, TetroEnum.J);
    const store = mockStore(initialState);
    const withProvider = (
      <Provider store={store}>
        <BoardContainer />
      </Provider>
    );
    const component = renderer.create(withProvider);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
