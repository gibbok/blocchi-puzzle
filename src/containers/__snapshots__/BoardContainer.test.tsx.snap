// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardContainer /> should render correctly 1`] = `
<div
  className="sc-AxmLO iQoDvo"
>
  <div
    className="sc-Axmtr bFEGpo"
  >
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fUKIku"
    >
      <div
        className="sc-AxjAm hTpnSM"
      >
        <div
          className="sc-AxirZ cTgbdZ"
        />
        <div
          className="sc-AxiKw dglFec"
        />
        <div
          className="sc-AxhCb hRLSKp"
        />
        <div
          className="sc-AxhUy gUafog"
        />
        <div
          className="sc-AxgMl jUBJBs"
        />
      </div>
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fUKIku"
    >
      <div
        className="sc-AxjAm hTpnSM"
      >
        <div
          className="sc-AxirZ cTgbdZ"
        />
        <div
          className="sc-AxiKw dglFec"
        />
        <div
          className="sc-AxhCb hRLSKp"
        />
        <div
          className="sc-AxhUy gUafog"
        />
        <div
          className="sc-AxgMl jUBJBs"
        />
      </div>
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fUKIku"
    >
      <div
        className="sc-AxjAm hTpnSM"
      >
        <div
          className="sc-AxirZ cTgbdZ"
        />
        <div
          className="sc-AxiKw dglFec"
        />
        <div
          className="sc-AxhCb hRLSKp"
        />
        <div
          className="sc-AxhUy gUafog"
        />
        <div
          className="sc-AxgMl jUBJBs"
        />
      </div>
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fUKIku"
    >
      <div
        className="sc-AxjAm hTpnSM"
      >
        <div
          className="sc-AxirZ cTgbdZ"
        />
        <div
          className="sc-AxiKw dglFec"
        />
        <div
          className="sc-AxhCb hRLSKp"
        />
        <div
          className="sc-AxhUy gUafog"
        />
        <div
          className="sc-AxgMl jUBJBs"
        />
      </div>
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
    <div
      className="sc-AxheI fwUCIC"
    >
      <div
        className="sc-AxjAm hQvQLG"
      />
    </div>
  </div>
</div>
`;
